### DISCUSSION TITLE
HomeAssistant timing out on /api/stats

### DISCUSSION THREAD

**Discussion by vajonam** [blakeblackshear/frigate#6383] [Ask A Question] [answered]
I see this error `Timeout error fetching information from http://192.168.100.2:5000/api/stats` about 50 times a day, hitting in a browser takes a second or two to come back, which is the same time it takes to refresh the System page on the UI. is this a know issue?

**Comment by NickM-27** [comment]
I mean, I definitely don't see any timeouts myself. Taking 2+ seconds is expected given that top, nvidia-smi, etc. are being run and these are not instant

**Comment by vajonam** [comment]
Hmm. okay I will keep an eye on it. I will try to isolate in my environment. yeah understand that stats calls is expensive, the fact that HA is constantly polling that endpoint a place to optimize? maybe push stats to HA once in 4-5 minutes? 

```
frigate           | 2023-05-04 14:53:40.418749056  10.1.1.136 - - [04/May/2023:14:53:40 -0400] "GET /api/stats HTTP/1.1" 200 3700 "-" "HomeAssistant/2023.5.0 aiohttp/3.8.4 Python/3.10" "-"
frigate           | 2023-05-04 14:53:48.436482521  10.1.1.136 - - [04/May/2023:14:53:48 -0400] "GET /api/stats HTTP/1.1" 200 3698 "-" "HomeAssistant/2023.5.0 aiohttp/3.8.4 Python/3.10" "-"
frigate           | 2023-05-04 14:53:56.423995443  10.1.1.136 - - [04/May/2023:14:53:56 -0400] "GET /api/stats HTTP/1.1" 200 3731 "-" "HomeAssistant/2023.5.0 aiohttp/3.8.4 Python/3.10" "-"
frigate           | 2023-05-04 14:54:04.414609150  10.1.1.136 - - [04/May/2023:14:54:04 -0400] "GET /api/stats HTTP/1.1" 200 3697 "-" "HomeAssistant/2023.5.0 aiohttp/3.8.4 Python/3.10" "-"
frigate           | 2023-05-04 14:54:12.433136927  10.1.1.136 - - [04/May/2023:14:54:12 -0400] "GET /api/stats HTTP/1.1" 200 3698 "-" "HomeAssistant/2023.5.0 aiohttp/3.8.4 Python/3.10" "-"
frigate           | 2023-05-04 14:54:20.435836099  10.1.1.136 - - [04/May/2023:14:54:20 -0400] "GET /api/stats HTTP/1.1" 200 3698 "-" "HomeAssistant/2023.5.0 aiohttp/3.8.4 Python/3.10" "-"
frigate           | 2023-05-04 14:54:28.425212636  10.1.1.136 - - [04/May/2023:14:54:28 -0400] "GET /api/stats HTTP/1.1" 200 3731 "-" "HomeAssistant/2023.5.0 aiohttp/3.8.4 Python/3.10" "-"
frigate           | 2023-05-04 14:54:36.410817415  10.1.1.136 - - [04/May/2023:14:54:36 -0400] "GET /api/stats HTTP/1.1" 200 3698 "-" "HomeAssistant/2023.5.0 aiohttp/3.8.4 Python/3.10" "-"
```

**Reply by NickM-27** [reply, marked as answer]
I mean, it uses the stats for many of its sensors, so 4-5 minutes means the graphs in HA for CPU / GPU / camera fps, etc. usage would be very delayed and not useful. an automation that receives the signal for a down camera stream would like 4-5 minutes to activate

**Reply by vajonam** [reply]
Right. I didn't know what all the /tats endpoint was used for or maybe use mqtt to push back stream/stats as they change. Will leave that up to you all. In the meanwhile i will see if I can figure out the reason for the timeouts on HA.
